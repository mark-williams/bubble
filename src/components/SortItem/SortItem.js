import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const getItemBackgroundColour = (n) => {
  const color = (n * 20) % 360;
  return `hsl(${color}, 40%, 50%)`;
};

const StyledItem = styled.div`
 width: 50px;
  border: solid 1px #222;
  text-align: center;
  padding: 0.8em 1em;
  margin: 0.1em;
  background-color: ${props => getItemBackgroundColour(props.item)};
  color: #fff;
`;

const SortItem = props => (
  <StyledItem {...props}>{props.item}</StyledItem>
);

SortItem.propTypes = {
  item: PropTypes.number,
};

export default SortItem;
