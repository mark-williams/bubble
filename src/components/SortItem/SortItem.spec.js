import React from 'react';
import renderer from 'react-test-renderer';
import SortItem from './SortItem';

describe('SortItem', () => {
  it('snapshot', () => {
    const tree = renderer.create(
      <SortItem item={171} />,
    );
    expect(tree).toMatchSnapshot();
  });
});
