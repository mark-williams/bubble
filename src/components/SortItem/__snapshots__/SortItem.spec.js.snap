// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortItem snapshot 1`] = `
<div
  className="sortItem"
  style={
    Object {
      "backgroundColor": "hsl(180, 40%, 50%)",
    }
  }
>
  171
</div>
`;
